### PR TITLE
chore: make create_plan a public API

### PR DESCRIPTION
**Pull request by eddyxu**
Expose planning as public API for third party.

### CODE REVIEW

File: rust/lance/src/io/exec/knn.rs [hunk 1]
```diff
@@ -140,11 +140,11 @@ impl DFRecordBatchStream for KNNFlatStream {
 /// - `input` schema does not have "_distance" column.
 #[derive(Debug)]
 pub struct KNNFlatExec {
-    /// Input node.
-    input: Arc<dyn ExecutionPlan>,
+    /// Inner input node.
```
Comment (westonpace) [inline rust/lance/src/io/exec/knn.rs:143]: What does "inner" mean here?

**Review by westonpace — APPROVED** [review]
Why physical plan and not logical plan?  I guess we don't really have logical equivalents for our nodes?

File: rust/lance/src/io/exec/knn.rs (hunk 1 above)
Comment (eddyxu) [reply rust/lance/src/io/exec/knn.rs:143]: As input source?

**Comment by eddyxu** [comment]
Ideally it'd be better to have logic plan to be exposed.  Do we plan to do it soon?
We are not currently using LogicalPlan

**Comment by wjones127** [comment]
> Ideally it'd be better to have logic plan to be exposed. Do we plan to do it soon?

Yes, I'll likely do that shortly after refactoring flat search.

See: https://github.com/lancedb/lance/issues/1782

**Pull request merged**